src/Stack/MetricsCollector.php

```diff
@@ -0,0 +1,229 @@
+<?php
+namespace Stack;
+
+class MetricsRegistry
+{
+    public function __construct($metrics = [])
+    {
+        $storage = new \Prometheus\Storage\APC();
+        $this->registry = new \Prometheus\CollectorRegistry($storage);
+        $this->register_metrics($metrics);
+    }
+
+    public function register_metrics($metrics = [])
+    {
+        foreach ($metrics as $metric => [$kind, $description, $labels]) {
+            [$namespace, $name] = explode('.', $metric);
+
+            switch ($kind) {
+                case 'histogram':
+                    $this->metrics[$metric] = $this->registry->registerHistogram(
+                        $namespace,
+                        $name,
+                        $description,
+                        $labels
+                    );
+                    break;
+
+                case 'counter':
+                    $this->metrics[$metric] = $this->registry->registerCounter(
+                        $namespace,
+                        $name,
+                        $description,
+                        $labels
+                    );
+                    break;
+
+                case 'gauge':
+                    $this->metrics[$metric] = $this->registry->registerGauge(
+                        $namespace,
+                        $name,
+                        $description,
+                        $labels
+                    );
+                    break;
+            }
+        }
+    }
+
+    public function get_gauge($metric)
+    {
+        return $this->metrics[$metric];
+    }
+
+    public function get_counter($metric)
+    {
+        return $this->metrics[$metric];
+    }
+
+    public function get_histogram($metric)
```

Comment:
It looks like a metric can't have multiple representation types. Maybe just `getMetric` should do the trick.

Reply:
The reason they have different names is that the call signature for collecting a metric depends on the metric type:
```
$this->metrics->get_counter('wp.requests')->incBy(1, [$request_type]);
$this->metrics->get_histogram('wp.peak_memory')->observe($peak_memory, [$request_type]);
```
https://github.com/presslabs/stack-mu-plugin/pull/7/files#diff-c53d875ceab7ea299db2d8883237e576R121-R122

Reply:
Yeap, saw that, but it felt like a forced abstraction or a force typed hinting.

Reply:
Poor man's tools in lack of better options.
I *could* create a `Metric` class to abstract the type away, and just use `set`, but it feels a bit over-engineered and a bit magic (hiding the type of metric)... I don't know...